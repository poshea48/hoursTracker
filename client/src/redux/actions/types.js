export const GET_ERRORS = "GET_ERRORS";
export const SET_CURRENT_USER = "SET_CURRENT_USER";

export const CHART_LOADING = "CHART_LOADING";
export const GET_DAILY = "GET_DAILY";
export const GET_WEEKLY = "GET_WEEKLY";
export const GET_MONTHLY = "GET_MONTHLY";

export const START_TIMER = "START_TIMER";
export const STOP_TIMER = "STOP_TIMER";
export const RESET_TIMER = "RESET_TIMER";
export const LOG_HOURS = "LOG_HOURS";
export const UPDATE_TIMER = "UPDATE_TIMER";

export const HOURS_ARCHIVED = "HOURS_ARCHIVED";
