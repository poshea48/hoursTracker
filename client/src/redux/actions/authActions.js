import axios from 'axios';
import { GET_ERRORS, SET_CURRENT_USER } from './types';
import setAuthToken from '../../utils/setAuthToken';
import jwt_decode from 'jwt-decode'

export const registerUser = (userData, history) => dispatch => {
  axios
    .post('/api/users/register', userData)
    .then(res => history.push('/login'))
    .catch(err =>
      dispatch({
        type: GET_ERRORS,
        payload: err.response.data
      })
    )
  return {
    type: GET_ERRORS,
    payload: userData
  }

}

// login - get user token
export const loginUser = userData => dispatch => {
  axios
    .post('/api/users/login', userData)
    .then(res => {
      // save token in local storage
      const { token } = res.data
      localStorage.setItem('jwtTokenHoursTracker', token)
      // set token to auth header
      setAuthToken(token);
      // decode token to get user data
      const decoded = jwt_decode(token)
      dispatch(setCurrentUser(decoded))
    })
    .catch(err =>
      dispatch({
        type: GET_ERRORS,
        payload: err.response.data
      })
    )
}

//log user out
export const logoutUser = () => dispatch => {
  //Remove token from localStorage
  localStorage.removeItem('jwtTokenHoursTracker')
  setAuthToken(false)
  dispatch(setCurrentUser({}))
}

export const setCurrentUser = decoded => {
  return {
    type: SET_CURRENT_USER,
    payload: decoded
  }
}
