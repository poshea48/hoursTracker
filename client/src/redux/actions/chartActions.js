import axios from "axios";
import {
  GET_DAILY,
  GET_WEEKLY,
  GET_MONTHLY,
  CHART_LOADING,
  GET_ALL_PROJECTS,
  GET_DAILY_PROJECT
} from "./types";
import getDateForDb from "../../utils/getDateForDb";

export const getDailyChart = (hoursToday, dateToday) => dispatch => {
  dispatch(setChartLoading());
  const date = getDateForDb(dateToday);
  axios
    .get(`/api/hours/daily`, {
      params: {
        today: date
      }
    })
    .then(res => {
      const dailyData = [...res.data];
      dailyData[dailyData.length - 1].hours += Number(hoursToday);
      return dispatch({
        type: GET_DAILY,
        payload: res.data
      });
    })
    .catch(err =>
      dispatch({
        type: GET_DAILY,
        payload: {}
      })
    );
};

export const updateTodaysData = (data, hoursToday) => dispatch => {
  const newData = [...data];
  newData[newData.length - 1].hours = hoursToday;
  dispatch({
    type: GET_DAILY,
    payload: newData
  });
};

export const getWeeklyChart = () => dispatch => {
  dispatch(setChartLoading());
  axios
    .get("/api/hours/weekly")
    .then(res =>
      dispatch({
        type: GET_WEEKLY,
        payload: res.data.slice(Math.max(res.data.length - 5, 1))
      })
    )
    .catch(err =>
      dispatch({
        type: GET_WEEKLY,
        payload: null
      })
    );
};

export const getMonthlyChart = () => dispatch => {
  dispatch(setChartLoading());
  axios
    .get("/api/hours/monthly")
    .then(res =>
      dispatch({
        type: GET_MONTHLY,
        payload: res.data
      })
    )
    .catch(err =>
      dispatch({
        type: GET_MONTHLY,
        payload: null
      })
    );
};

const setChartLoading = () => {
  return {
    type: CHART_LOADING
  };
};

export const getDailyChartForProject = project => dispatch => {
  dispatch(setChartLoading());
  const { id, hoursToday, dateToday } = project;
  axios
    .get(`/api/hours/project/daily/${id}`, { params: { today: dateToday } })
    .then(res => {
      const dailyData = [...res.data];
      dailyData[dailyData.length - 1].hours += Number(hoursToday);
      return dispatch({
        type: GET_DAILY_PROJECT,
        payload: res.data
      });
    })
    .catch(err => console.log("error in getDailyChartForProject"));
};

export const getAllProjects = () => dispatch => {
  axios
    .get("/api/hours/projects")
    .then(res => {
      return dispatch({
        type: GET_ALL_PROJECTS,
        payload: res.data
      });
    })
    .catch(err =>
      dispatch({
        type: GET_ALL_PROJECTS,
        payload: []
      })
    );
};

// export const changeProject = projectName => dispatch => {
//   return dispatch({
//     type: CHANGE_PROJECT,
//     payload: projectName
//   });
// };

// const addProject = projectData => dispatch => {
//   axios
//     .post("/api/hours/add_project", projectData)
//     .then(res =>
//       dispatch({
//         type: GET_PROJECT,
//         payload: res.data
//       })
//     )
//     .catch(err =>
//       dispatch({
//         type: GET_PROJECT,
//         payload: null
//       })
//     );
// };
