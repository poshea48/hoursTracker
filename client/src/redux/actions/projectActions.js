import axios from "axios";
import { GET_PROJECT, REMOVE_PROJECT } from "./types";
import getDateForDb from "../../utils/getDateForDb";

export const getProject = projectName => dispatch => {
  axios
    .get("/api/hours/project/details", { params: { projectName } })
    .then(res => {
      let projectData = res.data[0];
      projectData.dateToday = getDateForDb();
      return dispatch({
        type: GET_PROJECT,
        payload: {
          active: true,
          id: projectData.id,
          name: projectData.name,
          totalHours: projectData.total_hrs,
          hoursToday: projectData.hours_today
        }
      });
    })
    .catch(err => console.log(err));
};

export const removeProject = () => dispatch => {
  return dispatch({
    type: REMOVE_PROJECT
  });
};

// export const changeProject = projectName => dispatch => {
//   return dispatch({
//     type: CHANGE_PROJECT,
//     payload: projectName
//   });
// };
