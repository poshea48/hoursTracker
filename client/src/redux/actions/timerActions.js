import axios from "axios";
import {
  START_TIMER,
  STOP_TIMER,
  RESET_TIMER,
  UPDATE_TIMER,
  HOURS_ARCHIVED
} from "./types";
import getTodaysDate from "../../utils/getTodaysDate";

export const startTimer = () => dispatch => {
  const today = getTodaysDate();
  dispatch({
    type: START_TIMER,
    payload: { time: new Date().getTime(), date: today }
  });
};

export const stopTimer = hoursToday => dispatch => {
  dispatch({
    type: STOP_TIMER,
    payload: Number(hoursToday.toFixed(3))
  });
};

export const updateTimer = ({
  hoursToday,
  dateToday,
  startTime
}) => dispatch => {
  hoursToday = Number(hoursToday);
  startTime = Number(startTime);
  const today = new Date();
  let forced = today.toDateString() !== new Date(dateToday).toDateString();
  return dispatch({
    type: UPDATE_TIMER,
    payload: { dateToday, hoursToday, startTime, forced }
  });
};

export const resetTimer = () => dispatch => {
  dispatch({
    type: RESET_TIMER
  });
};

export const logHours = (hours, date) => dispatch => {
  axios
    .post("/api/hours/log-hours", { hours, date })
    .then(res =>
      dispatch({
        type: RESET_TIMER
      })
    )
    .catch(err => console.log(err));
};

export const archiveHours = () => dispatch => {
  console.log("archiving...");
  axios
    .post("/api/hours/archive-hours")
    .then(res => {
      return dispatch({ type: HOURS_ARCHIVED });
    })
    .catch(err => console.log("hours not archived, something happened"));
};
