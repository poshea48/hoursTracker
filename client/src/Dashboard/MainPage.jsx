import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import Spinner from "../components/common/Spinner";
import { connect } from "react-redux";
import Layout from "../components/layout/Layout";
import Header from "../components/layout/Header";
import getTodaysDate from "../utils/getTodaysDate";
import ButtonNav from "../components/buttons/ButtonNav";
import DisplayGraph from "../components/charts/DisplayGraph";
import ChartNavigation from "../components/chartNavigation/ChartNavigation";
import {
  getDailyChart,
  getWeeklyChart,
  getMonthlyChart,
  updateTodaysData
} from "../redux/actions/chartActions";
import {
  startTimer,
  stopTimer,
  resetTimer,
  logHours,
  updateTimer,
  archiveHours
} from "../redux/actions/timerActions";
import { logoutUser } from "../redux/actions/authActions";

class MainPage extends PureComponent {
  handleStartClick = e => {
    e.preventDefault();
    let dateToday =
      localStorage.getItem("dateToday") ||
      this.props.timer.dateToday ||
      getTodaysDate();
    let startTime = new Date().getTime();
    localStorage.setItem("startTime", startTime);
    localStorage.setItem("dateToday", dateToday);
    this.props.startTimer(startTime);
  };

  handleStopClick = e => {
    if (this.props.timer.startTime === 0) {
      return;
    }
    let end = new Date().getTime();
    // adding seconds to hoursToday
    let hoursToday = Number(this.props.timer.hoursToday) || 0;
    hoursToday += (end - this.props.timer.startTime) / 3600000;
    localStorage.setItem("startTime", 0);
    localStorage.setItem("hoursToday", hoursToday.toFixed(3));
    this.props.stopTimer(hoursToday);
  };

  handleResetClick = e => {
    localStorage.setItem("hoursToday", 0);
    localStorage.setItem("dateToday", "");
    this.props.resetTimer();
  };

  handleLogClick = () => {
    const { hoursToday, dateToday, startTime } = this.props.timer;
    let addedHours = 0;
    if (startTime > 0) {
      addedHours = new Date().getTime() - startTime;
    }
    localStorage.setItem("hoursToday", 0);
    localStorage.setItem("dateToday", "");
    localStorage.setItem("startTime", 0);

    this.props.logHours(hoursToday + addedHours, dateToday);
  };

  // moved to ChartNavigation component
  // handleChartSelect = e => {
  //   if (e.target.value === "weekly") {
  //     this.props.getWeeklyChart();
  //   } else if (e.target.value === "monthly") {
  //     this.props.getMonthlyChart();
  //   } else {
  //     const { hoursToday, dateToday } = this.props.timer;
  //     this.props.getDailyChart(hoursToday, dateToday);
  //   }
  // };

  getDataFromLocal = () => {
    if (
      !localStorage.getItem("dateToday") ||
      localStorage.getItem("dateToday") === ""
    ) {
      localStorage.setItem("dateToday", getTodaysDate());
    }

    if (!localStorage.getItem("hoursToday")) {
      localStorage.setItem("hoursToday", "0");
    }

    if (!localStorage.getItem("startTime")) {
      localStorage.setItem("startTime", "0");
    }
    const { dateToday, hoursToday, startTime } = localStorage;

    return { dateToday, hoursToday, startTime };
  };

  componentDidMount() {
    const localData = this.getDataFromLocal();

    this.props.getDailyChart(localData.hoursToday, localData.dateToday);
    this.props.updateTimer(localData);

    // Archive hours
    // TODO => setup cron job
    if (new Date(localData.dateToday).getDate() === 7) {
      return this.props.timer.archived ? null : this.props.archiveHours();
    }
  }

  // add in shouldComponentUpdate for hoursToday

  // Check if hoursToday in timer object changed, if so then update daily chart
  componentDidUpdate(prevProps) {
    if (!this.props.timer.dateToday) {
      const localData = this.getDataFromLocal();
      this.props.getDailyChart(localData.hoursToday, localData.dateToday);
      return this.props.updateTimer(localData);
    }
    if (
      prevProps.timer.hoursToday !== this.props.timer.hoursToday &&
      this.props.chart.chartType === "daily"
    ) {
      const { hoursToday, dateToday } = this.props.timer;
      return this.props.getDailyChart(hoursToday, dateToday);
    }

    if (prevProps.project.active && !this.props.project.active) {
      const { hoursToday, dateToday } = this.props.timer;

      return this.props.getDailyChart(hoursToday, dateToday);
    }
  }

  render() {
    // const decoded = jwt_decode(localStorage.jwtTokenHoursTracker);
    // const currentTime = Date.now() / 1000;
    const { start, stop, reset, log } = this.props.timer.disabled;
    const { chartType, data, loading } = this.props.chart;
    const { project } = this.props;
    const actions = {
      Start: {
        id: "startId",
        action: this.handleStartClick,
        disabled: start
      },
      Stop: {
        id: "stopId",
        action: this.handleStopClick,
        disabled: stop
      },
      Reset: {
        id: "resetId",
        action: this.handleResetClick,
        disabled: reset
      },
      "Log Hours": {
        id: "logId",
        action: this.handleLogClick,
        disabled: log
      }
    };
    return (
      <Layout auth={this.props.auth}>
        <Header />
        <ButtonNav
          actions={actions}
          startDisabled={start}
          stopDisabled={stop}
          resetDisabled={reset}
          logDisabled={log}
        />

        <ChartNavigation
          chartType={chartType}
          project={project.name || "total"}
          handleSelect={this.handleChartSelect}
        />

        {loading || typeof data === "undefined" ? (
          <Spinner />
        ) : (
          <DisplayGraph chartType={chartType} data={data} />
        )}
      </Layout>
    );
  }
}

MainPage.propTypes = {
  getDailyChart: PropTypes.func.isRequired,
  getWeeklyChart: PropTypes.func.isRequired,
  getMonthlyChart: PropTypes.func.isRequired,
  updateTodaysData: PropTypes.func.isRequired,
  updateTimer: PropTypes.func.isRequired,
  startTimer: PropTypes.func.isRequired,
  stopTimer: PropTypes.func.isRequired,
  resetTimer: PropTypes.func.isRequired,
  logHours: PropTypes.func.isRequired,
  archiveHours: PropTypes.func.isRequired,
  logoutUser: PropTypes.func.isRequired,
  chart: PropTypes.object.isRequired,
  timer: PropTypes.object.isRequired,
  project: PropTypes.object.isRequired
};

const mapStateToProps = state => {
  return {
    auth: state.auth,
    chart: state.chart,
    timer: state.timer,
    errors: state.errors,
    project: state.project
  };
};

export default connect(
  mapStateToProps,
  {
    getDailyChart,
    getWeeklyChart,
    getMonthlyChart,
    updateTodaysData,
    startTimer,
    stopTimer,
    resetTimer,
    logHours,
    updateTimer,
    logoutUser,
    archiveHours
  }
)(MainPage);
