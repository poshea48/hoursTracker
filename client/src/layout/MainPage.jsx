import React, { Component } from "react";
import PropTypes from "prop-types";
import Spinner from "../common/Spinner";
import { connect } from "react-redux";
import FlashMessage from "./FlashMessage";
import Header from "./Header";
import getTodaysDate from "../utils/getTodaysDate";
import ButtonNav from "../buttons/ButtonNav";
import Navbar from "./Navbar/Navbar";
import Graph from "../charts/Graph";
import NavHistory from "./NavHistory";
import { getDailyChart, updateTodaysData } from "../redux/actions/chartActions";
import {
  startTimer,
  stopTimer,
  resetTimer,
  logHours,
  updateTimer,
  archiveHours
} from "../redux/actions/timerActions";
import { logoutUser } from "../redux/actions/authActions";

import jwt_decode from "jwt-decode";
import setAuthToken from "../utils/setAuthToken";

class MainPage extends Component {
  state = {
    flash: ""
  };

  onLogoutClick = e => {
    let message = "Logout has been disabled";
    this.setState({ flash: message });
    return;
    e && e.preventDefault();
    const { hoursToday } = this.props.timer;
    if (hoursToday > 0) {
      this.props.logHours(hoursToday);
    }
    ["hoursToday", "startTime", "dateToday"].forEach(i =>
      localStorage.removeItem(i)
    );
    console.log("should be logging out");
    this.props.logoutUser();
  };

  handleStartClick = e => {
    let dateToday =
      localStorage.getItem("dateToday") ||
      this.props.timer.dateToday ||
      getTodaysDate();
    let startTime = new Date().getTime();
    localStorage.setItem("startTime", startTime);
    localStorage.setItem("dateToday", dateToday);
    this.props.startTimer(startTime);
  };

  handleStopClick = e => {
    if (this.props.timer.startTime === 0) {
      return;
    }
    let end = new Date().getTime();
    // adding seconds to hoursToday
    let hoursToday = Number(this.props.timer.hoursToday) || 0;
    hoursToday += (end - this.props.timer.startTime) / 3600000;
    localStorage.setItem("startTime", 0);
    localStorage.setItem("hoursToday", hoursToday.toFixed(3));
    this.props.stopTimer(hoursToday);
  };

  handleResetClick = e => {
    localStorage.setItem("hoursToday", 0);
    localStorage.setItem("dateToday", "");
    this.props.resetTimer();
  };

  handleLogClick = () => {
    const { hoursToday, dateToday, startTime } = this.props.timer;
    let addedHours = 0;
    if (startTime > 0) {
      addedHours = new Date().getTime() - startTime;
    }
    localStorage.setItem("hoursToday", 0);
    localStorage.setItem("dateToday", "");
    localStorage.setItem("startTime", 0);

    this.props.logHours(hoursToday + addedHours, dateToday);
  };

  removeFlashMessage = e => {
    e.preventDefault();
    this.setState({ flash: "" });
  };

  getDataFromLocal = () => {
    if (
      !localStorage.getItem("dateToday") ||
      localStorage.getItem("dateToday") === ""
    ) {
      localStorage.setItem("dateToday", getTodaysDate());
    }

    if (!localStorage.getItem("hoursToday")) {
      localStorage.setItem("hoursToday", "0");
    }

    if (!localStorage.getItem("startTime")) {
      localStorage.setItem("startTime", "0");
    }
    const { dateToday, hoursToday, startTime } = localStorage;

    return { dateToday, hoursToday, startTime };
  };

  componentDidMount() {
    console.log("mounting");
    const localData = this.getDataFromLocal();

    this.props.getDailyChart(localData.hoursToday, localData.dateToday);
    this.props.updateTimer(localData);

    // Archive hours
    if (new Date(localData.dateToday).getDate() === 7) {
      return this.props.timer.archived ? null : this.props.archiveHours();
    }
  }

  // Check if hoursToday in timer object changed, if so then update daily chart
  componentDidUpdate(prevProps) {
    console.log("updating");
    // disabling login/logout
    // if (localStorage.jwtTokenHoursTracker) {
    //   setAuthToken(localStorage.jwtTokenHoursTracker);
    //   const decoded = jwt_decode(localStorage.jwtTokenHoursTracker);
    //
    //   // Check for expired token
    //   const currentTime = Date.now() / 1000;
    //   if (decoded.exp < currentTime) {
    //     this.onLogoutClick();
    //   }
    // }

    if (!this.props.timer.dateToday) {
      const localData = this.getDataFromLocal();
      console.log(localData);
      this.props.getDailyChart(localData.hoursToday, localData.dateToday);
      return this.props.updateTimer(localData);
    }
    if (
      prevProps.timer.hoursToday !== this.props.timer.hoursToday &&
      this.props.chart.chartType === "daily"
    ) {
      const { hoursToday, dateToday } = this.props.timer;
      return this.props.getDailyChart(hoursToday, dateToday);
    }
  }

  render() {
    // const decoded = jwt_decode(localStorage.jwtTokenHoursTracker);
    // const currentTime = Date.now() / 1000;
    const { start, stop, reset, log } = this.props.timer.disabled;
    const { chartType, data, loading } = this.props.chart;
    const actions = {
      Start: {
        id: "startId",
        action: this.handleStartClick,
        disabled: start
      },
      Stop: {
        id: "stopId",
        action: this.handleStopClick,
        disabled: stop
      },
      Reset: {
        id: "resetId",
        action: this.handleResetClick,
        disabled: reset
      },
      "Log Hours": {
        id: "logId",
        action: this.handleLogClick,
        disabled: log
      }
    };
    return (
      <div>
        <div className="main">
          <Navbar onLogoutClick={this.onLogoutClick} auth={this.props.auth} />
          {this.state.flash && (
            <FlashMessage
              message={this.state.flash}
              remove={this.removeFlashMessage}
            />
          )}
          <Header />
          <ButtonNav
            actions={actions}
            startDisabled={start}
            stopDisabled={stop}
            resetDisabled={reset}
            logDisabled={log}
            startTimer={this.handleStartClick}
            stopTimer={this.handleStopClick}
            resetTimer={this.handleResetClick}
            logHours={this.handleLogClick}
          />

          <NavHistory chartType={chartType} />
          {loading || typeof data === "undefined" ? (
            <Spinner />
          ) : (
            <div className="chart">
              <Graph chartType={chartType} data={data} />
            </div>
          )}
        </div>
      </div>
    );
  }
}

MainPage.propTypes = {
  getDailyChart: PropTypes.func.isRequired,
  updateTodaysData: PropTypes.func.isRequired,
  updateTimer: PropTypes.func.isRequired,
  startTimer: PropTypes.func.isRequired,
  stopTimer: PropTypes.func.isRequired,
  resetTimer: PropTypes.func.isRequired,
  logHours: PropTypes.func.isRequired,
  archiveHours: PropTypes.func.isRequired,
  logoutUser: PropTypes.func.isRequired,
  chart: PropTypes.object.isRequired,
  timer: PropTypes.object.isRequired
};

const mapStateToProps = state => {
  return {
    auth: state.auth,
    chart: state.chart,
    timer: state.timer,
    errors: state.errors
  };
};

export default connect(
  mapStateToProps,
  {
    getDailyChart,
    updateTodaysData,
    startTimer,
    stopTimer,
    resetTimer,
    logHours,
    updateTimer,
    logoutUser,
    archiveHours
  }
)(MainPage);
