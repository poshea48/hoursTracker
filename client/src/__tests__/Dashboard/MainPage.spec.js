import React from "react";
import { shallow } from "enzyme";
import MainPage from "../../Dashboard/MainPage";
import * as timerActions from "../../redux/actions/timerActions";

describe("MainPage", () => {
  let wrapper;
  const props = {
    getDailyChart: jest.fn(),
    updateTodaysData: jest.fn(),
    updateTimer: jest.fn(),
    startTimer: jest.fn(),
    stopTimer: jest.fn(),
    resetTimer: jest.fn(),
    logHours: jest.fn(),
    archiveHours: jest.fn(),
    logoutUser: jest.fn(),
    chart: {
      type: "daily",
      data: {},
      loading: false
    },
    timer: {
      dateToday: "",
      hoursToday: 0,
      startTime: 0,
      forceLog: false,
      archived: false,
      disabled: {
        start: "false",
        stop: "false",
        reset: "false",
        log: "false"
      }
    }
  };
  beforeEach(() => (wrapper = shallow(<MainPage {...props} />)));

  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should disable appropriate buttons", () => {
    const mockEvent = {
      preventDefault: jest.fn()
    };

    // Mock actions we expect to be called

    expect(props.updateTimer.mock.calls.length).toBe(1);
    expect(props.getDailyChart.mock.calls.length).toBe(1);
  });
});
