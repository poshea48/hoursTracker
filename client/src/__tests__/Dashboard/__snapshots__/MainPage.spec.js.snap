// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(MainPage)
    archiveHours={[MockFunction]}
    chart={
      Object {
        "data": Object {},
        "loading": false,
        "type": "daily",
      }
    }
    getDailyChart={[MockFunction]}
    logHours={[MockFunction]}
    logoutUser={[MockFunction]}
    resetTimer={[MockFunction]}
    startTimer={[MockFunction]}
    stopTimer={[MockFunction]}
    timer={
      Object {
        "archived": false,
        "dateToday": "",
        "disabled": Object {
          "log": "false",
          "reset": "false",
          "start": "false",
          "stop": "false",
        },
        "forceLog": false,
        "hoursToday": 0,
        "startTime": 0,
      }
    }
    updateTimer={[MockFunction]}
    updateTodaysData={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
